### PR TITLE
add dotnet 6 target

### PR DESCRIPTION
**Pull request by sgentry**
What does this implement/fix? Explain your changes.
---------------------------------------------------
Adds support for targeting net6.0 and updates dependencies

Does this close any currently open issues?
------------------------------------------
No

Any relevant logs, error output, etc?
-------------------------------------
No

Where has this been tested?
---------------------------
**Operating System:** Windows 11

**.Net Core Verion:** net6.0

Does this introduce a breaking change?
-----------------------------------------

- [ ] Yes
- [x] No

Any other comments?
-------------------
Not 100% sure about updating `Microsoft.Extensions.Identity.Core` & `Microsoft.Extensions.Identity.Stores` from v2->v6 but prefer to use latest if not a breaking change. I've tested creating new user and login/logoff with sample application I'm working on. I can downgrade those if there's concerns.

### CODE REVIEW

**Comment by theranger** [comment]
Could we get this into mainline? Many projects cannot be migrated to .net6 because this package is not compatible with that version.

**Comment by jaytonic** [comment]
@matteofabbri Any chance you could accept this PR? I also need it for a project that runs on .Net 6

**Comment by Lukejkw** [comment]
We're using this package with .NET 6 without issue @theranger

File: src/AspNetCore.Identity.Mongo/AspNetCore.Identity.Mongo.csproj [hunk 1]
```diff
@@ -26,18 +26,18 @@
...
 	<ItemGroup Condition="$(TargetFramework) == 'netstandard2.1'">
-		<PackageReference Include="Microsoft.AspNetCore.Authentication" Version="2.1.2" />
-		<PackageReference Include="Microsoft.AspNetCore.Identity" Version="2.1.2" />
-		<PackageReference Include="Microsoft.Extensions.Identity.Core" Version="2.1.6" />
-		<PackageReference Include="Microsoft.Extensions.Identity.Stores" Version="2.1.6" />
+		<PackageReference Include="Microsoft.AspNetCore.Authentication" Version="2.2.0" />
+		<PackageReference Include="Microsoft.AspNetCore.Identity" Version="2.2.0" />
+		<PackageReference Include="Microsoft.Extensions.Identity.Core" Version="6.0.1" />
+		<PackageReference Include="Microsoft.Extensions.Identity.Stores" Version="6.0.1" />
 	</ItemGroup>
```
Comment (daviddesmet) [inline src/AspNetCore.Identity.Mongo/AspNetCore.Identity.Mongo.csproj:41]: This block is specific for .NET Standard 2.1, it shouldn't reference other versions that are not part of it.

**Review by daviddesmet — CHANGES_REQUESTED** [review]

File: src/AspNetCore.Identity.Mongo/AspNetCore.Identity.Mongo.csproj (hunk 1 above)
Comment (daviddesmet) [reply src/AspNetCore.Identity.Mongo/AspNetCore.Identity.Mongo.csproj:41]: In other words, that block shouldn't have changed.

**Pull request merged**